openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -78,15 +78,11 @@ def __init__(self, course_key, request, username=''):
             username or self.request.user.username,
             course_key,
         )
-        # We must compute course load access *before* setting up masquerading,
-        # else course staff (who are not enrolled) will not be able view
-        # their course from the perspective of a learner.
         self.load_access = check_course_access(
             self.overview,
             self.request.user,
             'load',
             check_if_enrolled=True,
-            check_survey_complete=False,
```

Comment:
I've re-enabled this check, because the frontend side of this PR will now support survey errors correctly. It seems like it was an anomoly that it was disabled in first place - I think just because they didn't have that survey error redirect done yet?